### DISCUSSION TITLE
How do I create a column based on the value of other columns in a selectable (to be used in a view)?

### DISCUSSION THREAD

**Discussion by mcarans** [sqlalchemy/sqlalchemy#12126] [Usage Questions] [answered]
I'm trying to do something like the following in which I create a column "admin_level" that is based on the values in columns "name" and "unspecified". (The selectable is used to generate a view.) How do I do this?
 
```
select(
        DBAdmin1.name.label("admin1_name"),
        DBAdmin1.code.label("admin1_code"),
        DBAdmin1.is_unspecified.label("admin1_is_unspecified"),
        case(
            (DBAdmin1.name is not None or DBAdmin1.is_unspecified is False, 1),
            else_=0
        ).label(
            "admin_level"
        ),
```

When I tried the above and looked at the SQL for the generated column, it was not what I intended:
```
        CASE
            WHEN true THEN 1
            ELSE 0
        END AS admin_level,
```

I guess it must have evaluated when compiling the SQL instead of when executing it. I wondered if I need something like a "column_property", but am not sure how to use that in a selectable. How do I make the column evaluate to 1 or 0 row by row based on the values in the columns "name" and "unspecified"?

**Comment by CaselIT** [comment, marked as answer]
Hi,

You can't use `or` / `and` like that when building expressions in python. Use the functions or_/and_ of sqlalchemy

**Reply by mcarans** [reply]
Thanks for your suggestion @CaselIT . I just tried it, but seems to evaluate to the same SQL as above:
```
        DBAdmin1.name.label("admin1_name"),
        DBAdmin1.code.label("admin1_code"),
        DBAdmin1.is_unspecified.label("admin1_is_unspecified"),
        case(
            (or_(DBAdmin1.name is not None, DBAdmin1.is_unspecified is False), 1),
            else_=0
        ).label(
            "admin_level"
        ),
```

**Reply by CaselIT** [reply]
Oh sorry. `is` and `is not` are also like and / or. Use the methods `.is_()` and `.is_not()`.

The issue is the same, can't override the operators, so you get back a boolean

**Reply by mcarans** [reply]
Thanks so much! Working now. My final case statement is something like this:
```
        case(
            (
                or_(
                    and_(
                        DBAdmin1.name.is_not(None),
                        DBAdmin1.name != "",
                    ),
                    DBAdmin1.is_unspecified.is_(False),
                ),
                1,
            ),
            else_=0,
        ).label("admin_level"),

```
